source/DD4T.Providers.Tridion9.5.CIL/TridionComponentPresentationProvider.cs

```diff
@@ -88,16 +88,24 @@ public string GetContent(string uri, string templateUri = "")
         /// <returns></returns>
         public List<string> GetContentMultiple(string[] componentUris)
         {
-//            TcmUri uri = new TcmUri(componentUris.First());
-            var components =
-                componentUris
-                .Select(componentUri => { TcmUri uri = new TcmUri(componentUri); return (T.ComponentPresentation)GetComponentPresentationFactory(uri.PublicationId).FindAllComponentPresentations(componentUri)[0]; })
-                .Where(cp => cp != null)
-                .Select(cp => cp.Content)
-                .ToList();
+            List<string> componentsContent = new List<string>();
 
-            return components;
+            foreach (var componentUri in componentUris)
+            {
+                TcmUri uri = new TcmUri(componentUri);
+                var componentPresentations = GetComponentPresentationFactory(uri.PublicationId).FindAllComponentPresentations(componentUri);
+
+                if (componentPresentations.Count > 0)
```

Comment:
.Any() is faster than .Count

Reply:
.Any() is not available on this IList variable.